ocaml/testsuite/tests/backtrace-multifiles/main.ml

```diff
@@ -0,0 +1,73 @@
+(* TEST_BELOW *)
+
+let test s =
+  let s' = (String.cat[@inlined never]) "! " s in
+  (Foo.h[@inlined]) s'; (* CR gbury: remove [@inlined] *)
```

Comment:
Presumably these CRs are to be addressed before merging?

Please also add a comment explaining what the test does, as it's kind of complicated to look at.

Reply:
Actually, these CR are because of #1199 , so I think it's better to merge this PR with them.

I'll add some documentation to the test to explain what it tests.

Reply:
I've added a comment at the end of `main.ml` to explain the goal of the test.